### DISCUSSION TITLE
MapInfo is null (Map file metadata)

### DISCUSSION THREAD

**Discussion by moiatto** [mapsforge/vtm#1047] [Q&A] [answered]
```
MapFileTileSource tileSource = new MapFileTileSource();
tileSource.setMapFileInputStream(fis);

// Get MAP supported languages
MapInfo mapInfo = tileSource.getMapInfo();
if (mapInfo != null)
    Log.d(TAG, "MapsForge: MAP info: " + mapInfo.languagesPreference);
```

`tileSource` is loaded correctly but `mapInfo` is always null.

I tried maps V5 ([maps](http://download.mapsforge.org/maps/v5/)):
albania.map
italy.map
andorra.map
cyprus.map

Any suggestions?

**Comment by devemux86** [comment, marked as answer]
Need to load the map to read its info.
See the [MapsforgeActivity](https://github.com/mapsforge/vtm/blob/master/vtm-android-example/src/org/oscim/android/test/MapsforgeActivity.java) example.

**Comment by moiatto** [comment]
Ah ok!
It doesn't provide the local language, right?

Thanks

**Reply by devemux86** [reply]
The [primary name](https://wiki.openstreetmap.org/wiki/Key:name) of the map features is always included.
The info provides the extra languages, as declared in map-writer.

**Comment by moiatto** [comment]
Understood

My intent is to check if the local language of the map matches the language of the device to set it otherwise set as default "en"
